tests/e2e/test_spyre_pc_scheduler_steps.py

```diff
@@ -465,6 +629,179 @@ def test_limit_blocks_no_prefix_hit(model: ModelInfo, backend: str,
         seeds=seeds)
 
 
+@pytest.mark.cpu
+@pytest.mark.chunked_prefill
+@pytest.mark.full_model
+@pytest.mark.prefix_caching
+# These values are all parameterized for test sorting
+@pytest.mark.parametrize("max_num_seqs", [4])
+@pytest.mark.parametrize("max_model_len", [256])
+@pytest.mark.parametrize("max_num_batched_tokens", [128])
+@pytest.mark.parametrize("available_blocks", [None])
+def test_double_prefix_hit_within_batch(model: ModelInfo, backend: str,
+                                        monkeypatch: pytest.MonkeyPatch,
+                                        set_random_seed, max_num_seqs: int,
+                                        max_model_len: int,
+                                        max_num_batched_tokens: int,
+                                        available_blocks: int):
+    """ Scenario where three equal and one different sequences are scheduled.
+    While prefilling the second and fourth sequence we have a prefix cache
+    hit and can reuse the first chunk. Note that the fetched prefix blocks 
+    are still part of the existing decode batch. Hence we have duplicated 
+    blocks in the block table for this example. More specifically, three
```

Comment:
>Note that the fetched prefix blocks 
    are still part of the existing decode batch. Hence we have duplicated 
    blocks in the block table for this example.
    
I'm a bit confused here, so we're identifying prefix cache hits but then we can't actually re-use the kv cache blocks at all, because the original sequence that filled those blocks is still running? So we're not actually caching anything in this case and we're recomputing?